### PR TITLE
allow bigger chunks of bind variables per SQL statement 

### PR DESCRIPTION
**Pull request by neurosis69**
I'd suggest to use parameter limits according to the SQLite version the full node is running with.
This is achieved using a dbwrapper constant and replacing the currently used constant  `MAX_SQLITE_PARAMETERS`.

To match the parameter name used  [in sqlite itself](https://www.sqlite.org/lang_expr.html#varparam), I also renamed the constant to `SQLITE_MAX_VARIABLE_NUMBER`.

### Performance impact
We should expect a slight improvement, also less parsing and syscalls.

This was tested using my ansible roles and the standard `long sync` operation on full node from main branch, starting from height 1537152 to 1556960. This should cover a short enough period with dust and regular tx in between.

System|Testcase|Runtime|%
---|---|---|---
RPi4B-8G|main|4h03m48s|100%
RPi4B-8G|PR|3h52m07s|95,21%

Test Limitation: Each sync test was only done 1x, variations can be high on such tests.

### Limits In SQLite:
SQLite versions < 3.32.0 allow up to 999 bind variables per single SQL statement
with versions >= 3.32.0 up to 32766 bind variables are possible
[Manual](https://www.sqlite.org/limits.html#max_variable_number)

### Identified enforced limits in chia

- minimal SQLite version for chia is 3.8
- in `full_node/coin_store.py` a constant `MAX_SQLITE_PARAMETERS = 900` is set and used to split parameter in chunks
- in `consensus/blockchain.py` 
  - the function `get_block_records_at` is using a default `batch_size: int = 900` and
  later in the function an `assert batch_size < 999`
  - `get_block_records_at` is calling the function `get_block_records_by_hash` in `full_node/block_store.py`
  - `get_block_records_by_hash` is not using `chia.util.chunks` right now, so not as easy to change

### CODE REVIEW

**Comment by mariano54** [comment]
from the manual:
the maximum value of a host parameter number is SQLITE_MAX_VARIABLE_NUMBER, which defaults to 999 for SQLite versions prior to 3.32.0 (2020-05-22) or 32766 for SQLite versions after 3.32.0. 
Which means it's just a default, we can set it to whatever we want right?

**Comment by neurosis69** [comment]
> Which means it's just a default, we can set it to whatever we want right? EDIT: or maybe we can only lower but not increase it?

We would need to recompile sqlite to raise the limits, if I understand it correctly.

When logging the sql commands during sync, then I only noticed a need for more bind variables per SQL statement for coin record inserts/removals. This may be different for wallet syncs though, didn't dive into that too much yet.

But I am currently also testing several other sqlite compile options which could have a positive impact with now drawback (hopefully). So I can just add a higher MAX_VARIABLE_NUMBER limit and do a testrun as well.

Using the following options during my tests, most of them taken from the [sqlite compile recommendations](https://www.sqlite.org/compile.html#recommended_compile_time_options).
```
 CFLAGS="-O2 -DSQLITE_ENABLE_STAT4=1 -DSQLITE_DEFAULT_MEMSTATUS -DSQLITE_LIKE_DOESNT_MATCH_BLOBS -DSQLITE_MAX_EXPR_DEPTH=0 -DSQLITE_OMIT_PROGRESS_CALLBACK -DSQLITE_OMIT_SHARED_CACHE -DSQLITE_USE_ALLOCA -DSQLITE_DIRECT_OVERFLOW_READ"
```

File: chia/util/db_wrapper.py [hunk 1]
```diff
@@ -7,6 +7,11 @@
 import aiosqlite
 
 
+if aiosqlite.sqlite_version_info < (3, 32, 0):
```
Comment (arvidn) [inline chia/util/db_wrapper.py:10]: I think ensuring that this comparison is correct is key to the correctness of this change

**Comment by arvidn** [comment]
right. the "default" mentioned in the manual is the default *limit* when building sqlite. We can't choose that limit, we have to live with it.

@neurosis69 what version of sqlite did you run that test with?

I think this change makes sense. I'm interested in running this on a more deterministic benchmark though.

**Comment by arvidn** [comment]
@neurosis69 would you mind running `pre-commit run --all` to fix up the patch?

File: chia/util/db_wrapper.py (hunk 1 above)
Comment (neurosis69) [reply chia/util/db_wrapper.py:10]: That's what I tested:

```python
>>> import aiosqlite
>>> aiosqlite.sqlite_version_info
(3, 34, 1)
>>> if aiosqlite.sqlite_version_info < (3, 32, 0):
...   SQLITE_MAX_VARIABLE_NUMBER = 999
... else:
...   SQLITE_MAX_VARIABLE_NUMBER = 32766
...
>>> print(SQLITE_MAX_VARIABLE_NUMBER)
32766
>>> if aiosqlite.sqlite_version_info < (3, 4, 0):
...   SQLITE_MAX_VARIABLE_NUMBER = 999
... else:
...   SQLITE_MAX_VARIABLE_NUMBER = 32766
...
>>> print(SQLITE_MAX_VARIABLE_NUMBER)
32766
>>> if aiosqlite.sqlite_version_info > (3, 32, 0):
...   SQLITE_MAX_VARIABLE_NUMBER = 999
... else:
...   SQLITE_MAX_VARIABLE_NUMBER = 32766
...
>>> print(SQLITE_MAX_VARIABLE_NUMBER)
999
```

**Comment by neurosis69** [comment]
> @neurosis69 what version of sqlite did you run that test with?

Tested with 3.34.1 and 3.38.5.

**Review by arvidn — APPROVED** [review]

**Pull request merged**